src/Illuminate/Support/Arr.php

```diff
@@ -62,19 +63,24 @@ public static function build($array, callable $callback)
     /**
      * Collapse an array of arrays into a single array.
      *
-     * @param  array  $array
+     * @param  \Traversable|array  $array
      * @return array
      */
     public static function collapse($array)
     {
+        // Iterate on the array for performance.
```

Comment:
Indeed, it would do the foreach using `ArrayIterator($this->items)`. But it's even slower than on a regular array (I've been surprised how slow this is).